### PR TITLE
Container-jersey2: Scala code replaced with Java

### PR DESCRIPTION
**Pull request by ollivir**

### CODE REVIEW

File: container-jersey2/src/main/java/com/yahoo/container/servlet/jersey/JerseyServletProvider.java [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        }
+        return ret;
+    }
+
+    private static Optional<String> detectResourceOrProvider(ClassLoader bundleClassLoader, String classEntry) {
+        try (InputStream inputStream = getResourceAsStream(bundleClassLoader, classEntry)) {
+            ResourceOrProviderClassVisitor visitor = ResourceOrProviderClassVisitor.visit(new ClassReader(inputStream));
+            return Optional.ofNullable(visitor.getClassName());
+        } catch (IOException e) {
+            // ignored
```
Comment (bjorncs) [inline container-jersey2/src/main/java/com/yahoo/container/servlet/jersey/JerseyServletProvider.java]: Should this exception be ignored?

File: container-jersey2/src/main/java/com/yahoo/container/servlet/jersey/ResourceOrProviderClassVisitor.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    private boolean isPublic = false;
+    private boolean isAbstract = false;
+
+    private boolean isInnerClass = false;
+    private boolean isStatic = false;
+
+    private boolean isAnnotated = false;
+
+    public ResourceOrProviderClassVisitor() {
+        super(Opcodes.ASM5);
```
Comment (bjorncs) [inline container-jersey2/src/main/java/com/yahoo/container/servlet/jersey/ResourceOrProviderClassVisitor.java]: Use ASM6 as in @gjoranv's recent PR: https://github.com/vespa-engine/vespa/pull/5919/files#diff-8.

**Review by bjorncs — CHANGES_REQUESTED** [review]

File: container-jersey2/src/main/java/com/yahoo/container/servlet/jersey/ResourceOrProviderClassVisitor.java (hunk 2 above)
Comment (bjorncs) [reply container-jersey2/src/main/java/com/yahoo/container/servlet/jersey/ResourceOrProviderClassVisitor.java]: EDIT: Noticed that the PR was not merged. Ignore above comment.

**Review by bjorncs — APPROVED** [review]

File: container-jersey2/src/main/java/com/yahoo/container/servlet/jersey/JerseyServletProvider.java [hunk 3]
```diff
@@ -67,9 +67,8 @@ private ResourceConfig resourceConfig(RestApiContext restApiContext) {
             ResourceOrProviderClassVisitor visitor = ResourceOrProviderClassVisitor.visit(new ClassReader(inputStream));
             return Optional.ofNullable(visitor.getClassName());
         } catch (IOException e) {
-            // ignored
+            throw new RuntimeException(e);
         }
-        return Optional.empty();
```
Comment (ollivir) [inline container-jersey2/src/main/java/com/yahoo/container/servlet/jersey/JerseyServletProvider.java]: The previous version here as questioned by @bjorncs was slightly different in behaviour than the Scala version. This change should bring it closer.

File: container-jersey2/src/main/java/com/yahoo/container/servlet/jersey/ComponentGraphProvider.java [hunk 4]
```diff
@@ -17,8 +17,7 @@
 import java.util.stream.Collectors;
 
 /**
- * Resolves jdisc container components for jersey 2 components. Similar to
- * Gjoran's ComponentGraphProvider for jersey 1.
+ * Resolves jdisc container components for jersey 2 components.
```
Comment (ollivir) [inline container-jersey2/src/main/java/com/yahoo/container/servlet/jersey/ComponentGraphProvider.java]: Jersey 1 comment wiped.

**Review by bjorncs — APPROVED** [review]

**Pull request merged**